packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/UpdateRulesTests.java

```diff
@@ -451,13 +451,40 @@ public void updateRuleWithConditionWithoutParamValueAndFail()
         rule.setConditions(conditions);
 
         restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()
-                .include(IS_SHARED)
                 .updateRule(rule.getId(), rule);
 
         restClient.assertStatusCodeIs(BAD_REQUEST);
         restClient.assertLastError().containsSummary("Parameter in condition must not be blank");
     }
 
+    /**
+     * Check we can update a rule by adding several actions.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES})
+    public void updateRuleAddActions()
+    {
+        final RestRuleModel rule = createAndSaveRule(createRuleModelWithModifiedValues());
+
+        STEP("Try to update the rule by adding several actions");
+        final Map<String, Serializable> copyParams =
+                Map.of("destination-folder", "dummy-folder-node", "deep-copy", true);
+        final RestActionBodyExecTemplateModel copyAction = createCustomActionModel("copy", copyParams);
+        final Map<String, Serializable> checkOutParams =
+                Map.of("destination-folder", "dummy-folder-node", "assoc-name", "cm:checkout", "assoc-type",
+                        "cm:contains");
+        final RestActionBodyExecTemplateModel checkOutAction = createCustomActionModel("check-out", checkOutParams);
+        final Map<String, Serializable> scriptParams = Map.of("script-ref", "dummy-script-node-id");
+        final RestActionBodyExecTemplateModel scriptAction = createCustomActionModel("script", scriptParams);
+        final RestRuleModel ruleModel = createRuleModelWithDefaultValues();
+        rule.setActions(Arrays.asList(copyAction, checkOutAction, scriptAction));
+        final RestRuleModel updatedRule = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()

```

Comment:
Since we have one big "Happy path" test for actions, I guess we can also add some of the "other options" to mix up with actions in the same test, if that makes sense.

Reply:
After our recent meeting and feedback from Stefan I added 2 new negative paths we found.
Anyway, I think that we should create a separate class for combined and thorough E2E REST tests of rules. 
I'd rather create a separate ticket for that as that could be a bigger effort which needs ideation and design work.